common/src/lib.rs

```diff
@@ -73,6 +73,22 @@ impl KeccakInclusionToDataRootProofOutput {
     }
 }
 
+pub struct PayyInclusionToDataRootProofInput {
+    pub data: Vec<u8>,
+    pub namespace_id: Namespace,
+    pub share_proofs: Vec<NamespaceProof>,
+    pub row_proof: RowProof,
+    pub data_root: [u8; 32],
+    // the ~8 UTXO hashes for a Payy block, bn254 scalar field elements represented as [u8; 32].
+    pub hashes: Vec<[u8; 32]>,
```

Comment:
I think now we need this to be 24 exactly, perhaps known & enforced at compile time, unsure if it will ever be different.
```suggestion
    // the ~8 UTXO hashes for a Payy block, bn254 scalar field elements represented as [u8; 32].
    pub hashes: [[u8; 32]; 24],
```

3 user proofs on UTXOs (that include 4 hashes) -> 2 `UTXO_agg` -> 1 `agg_agg` proof.
From https://excalidraw.com/#json=qq9tvy6-Z3i4LRw0aVPbN,8u_rJ0LFt9P9Rk-YDqb5GA